Makefile

```diff
@@ -20,14 +30,22 @@ remove-test-env:
 	@echo "Removing .env file"
 	@rm -f .env
 
-# Install additional dependencies to run VSCode testing in headless mode
-# ref: https://code.visualstudio.com/api/working-with-extensions/continuous-integration#github-actions
 .PHONY: test
-test: setup-test-env install-dependencies
-	sudo apt-get update
-	sudo apt install -y libgbm1 libgtk-3-0 xvfb
+test: setup-test-env install-test-dependencies install-dependencies
 	npx gulp ci
-	xvfb-run -a npx gulp test
+	@if [ $$(uname -s) = "Linux" ]; then \
+			xvfb-run -a npx gulp test; \
+	elif [ $$(uname -s) = "Darwin" ]; then \
+			if pgrep -x "Dock" > /dev/null; then \
+					echo "GUI session is active."; \
+					npx gulp test; \
+			else \
+					echo "No active GUI session. Aborting tests."; \
+					exit 1; \
+			fi \
```

Comment:
Darwin agents shouldn't even be re-attempted until we're able to get past this check.